at-http/src/main/java/io/engineblock/activitytypes/http/HttpActivity.java

```diff
@@ -0,0 +1,198 @@
+package io.engineblock.activitytypes.http;
+
+import activityconfig.ParsedStmt;
+import activityconfig.StatementsLoader;
+import activityconfig.yaml.StmtDef;
+import activityconfig.yaml.StmtsDocList;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import io.engineblock.activityapi.core.Activity;
+import io.engineblock.activityapi.core.ActivityDefObserver;
+import io.engineblock.activityimpl.ActivityDef;
+import io.engineblock.activityimpl.SimpleActivity;
+import io.engineblock.metrics.ActivityMetrics;
+import io.engineblock.util.TagFilter;
+import io.virtdata.api.ValuesBinder;
+import io.virtdata.core.Bindings;
+import io.virtdata.core.BindingsTemplate;
+import io.virtdata.core.ContextualBindingsTemplate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class HttpActivity extends SimpleActivity implements Activity, ActivityDefObserver {
+    private final static Logger logger = LoggerFactory.getLogger(HttpActivity.class);
+    private final ActivityDef activityDef;
+
+    public ContextualBindingsTemplate getContextualBindingsTemplate() {
+        return contextualBindingsTemplate;
+    }
+
+    private ContextualBindingsTemplate contextualBindingsTemplate;
+
+    public StmtsDocList getStmtDocList() {
+        return stmtDocList;
+    }
+
+    private final StmtsDocList stmtDocList;
+
+    private int stride;
+    private Integer maxTries;
+    private Boolean showstmnts;
+    public Timer bindTimer;
+    public Timer executeTimer;
+    public Histogram triesHisto;
+    public Timer resultTimer;
+    public Meter rowCounter;
+    public Histogram skippedTokens;
+    public Timer resultSuccessTimer;
+
+
+    public HttpActivity(ActivityDef activityDef) {
+        super(activityDef);
+
+        this.activityDef = activityDef;
+
+        String yaml_loc = activityDef.getParams().getOptionalString("yaml").orElse("default");
+
+        stmtDocList = StatementsLoader.load(logger,yaml_loc, "activities");
+    }
+
+
+
+    @Override
+    public void initActivity() {
+        super.initActivity();
+
+        stride = activityDef.getParams().getOptionalInteger("stride").orElse(1);
+
+        maxTries = activityDef.getParams().getOptionalInteger("maxTries").orElse(1);
+        showstmnts = activityDef.getParams().getOptionalBoolean("showstmnts").orElse(false);
+
+        contextualBindingsTemplate = createContextualBindingsTemplate();
+
```

Comment:
Somewhere in this method is where the sequencing needs to happen, as well as the conversion of binding templates to bindings.